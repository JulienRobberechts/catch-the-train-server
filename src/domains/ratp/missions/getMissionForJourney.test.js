const { getSchedulesForMissions } = require("./getMissionForJourney");
const each = require("jest-each").default;

const { missionsRepositoryMocked } = require("../../../repositories/ratp-api");

describe("SchedulesForMissions", () => {
  const UPAC1 = {
    destination: {
      name: "Cergy-Le-Haut",
      slug: "cergy+le+haut",
      letter: "U",
      order: 1,
      section: "A3",
    },
    missionCode: "UPAC",
    origin: {
      name: "Marne la Vallee-Chessy",
      slug: "marne+la+vallee+chessy",
      letter: "Q",
      order: 28,
      section: "A4",
    },
    _testCase:
      "Mission type Standard backward from Marne la Vallee-Chessy to Cergy-Le-Haut",
    type: 1,
    way: "backward",
  };

  const YOZZ2 = {
    destination: {
      name: "Rueil Malmaison",
      slug: "rueil+malmaison",
      letter: "Y",
      order: 7,
      section: "A1",
    },
    missionCode: "YOZZ",
    origin: {
      name: "Torcy",
      slug: "torcy",
      letter: "O",
      order: 25,
      section: "A4",
    },
    _testCase: "Mission type Special backward from Torcy to Rueil Malmaison",
    type: 2,
    way: "backward",
  };

  const TOZZ2 = {
    _testCase: "Mission type Special backward from Torcy to Poissy",
    destination: {
      name: "Poissy",
      slug: "poissy",
      letter: "T",
      order: 5,
      section: "A5",
    },
    missionCode: "TOZZ",
    origin: {
      name: "Torcy",
      slug: "torcy",
      letter: "O",
      order: 25,
      section: "A4",
    },
    type: 2,
    way: "backward",
  };

  const EXZZ2 = {
    _testCase: "Mission type Special forward from Vesinet to Joinville le Pont",
    destination: {
      name: "Joinville le Pont",
      slug: "joinville+le+pont",
      letter: "E",
      order: 20,
      section: "A2",
    },
    missionCode: "EXZZ",
    origin: {
      name: "Le Vesinet le Pecq",
      slug: "le+vesinet+le+pecq",
      letter: "X",
      order: 4,
      section: "A1",
    },
    type: 2,
    way: "forward",
  };

  const WEWX3 = {
    _testCase:
      "Mission type NonCommercial forward from Vesinet to Joinville le Pont",
    destination: {
      name: "Joinville le Pont",
      slug: "joinville+le+pont",
      letter: "E",
      order: 20,
      section: "A2",
    },
    missionCode: "WEWX",
    origin: {
      name: "Le Vesinet le Pecq",
      slug: "le+vesinet+le+pecq",
      letter: "X",
      order: 4,
      section: "A1",
    },
    type: 3,
    way: "forward",
  };

  describe("getSchedulesForMissions", () => {
    each([
      ["UPAC", UPAC1],
      ["YOZZ", YOZZ2],
      ["TOZZ", TOZZ2],
      ["EXZZ", EXZZ2],
    ]).test("should get schedule for missions %s", async (name, mission) => {
      const result = await getSchedulesForMissions(missionsRepositoryMocked, [
        mission,
      ]);
      expect({ mission, result }).toMatchSnapshot();
    });

    test("should NOT get schedule for missions non commercial missions", async () => {
      const invalidCall = async () => {
        await getSchedulesForMissions(missionsRepositoryMocked, [WEWX3]);
      };
      expect(invalidCall()).rejects.toEqual(
        new Error("The mission list contains non commercial missions!")
      );
    });
  });
});
