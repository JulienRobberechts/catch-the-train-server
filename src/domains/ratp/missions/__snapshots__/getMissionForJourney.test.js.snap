// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchedulesForMissions getSchedulesForMissions should get schedule for missions EXZZ 1`] = `
Object {
  "mission": Object {
    "_testCase": "Mission type Special forward from Vesinet to Joinville le Pont",
    "destination": Object {
      "letter": "E",
      "name": "Joinville le Pont",
      "order": 20,
      "section": "A2",
      "slug": "joinville+le+pont",
    },
    "missionCode": "EXZZ",
    "origin": Object {
      "letter": "X",
      "name": "Le Vesinet le Pecq",
      "order": 4,
      "section": "A1",
      "slug": "le+vesinet+le+pecq",
    },
    "type": 2,
    "way": "forward",
  },
  "result": Array [
    Object {
      "mission": "EXZZ",
      "stations": Array [
        "le+vesinet+le+pecq",
        "le+vesinet+centre",
        "chatou+croissy",
        "rueil+malmaison",
        "nanterre+ville",
        "nanterre+universite",
        "nanterre+prefecture",
        "grande+arche+la+defense",
        "charles+de+gaulle+etoile",
        "auber",
        "chatelet+les+halles",
        "gare+de+lyon",
        "nation",
        "vincennes",
        "fontenay+sous+bois",
        "nogent+sur+marne",
        "joinville+le+pont",
      ],
    },
  ],
}
`;

exports[`SchedulesForMissions getSchedulesForMissions should get schedule for missions TOZZ 1`] = `
Object {
  "mission": Object {
    "_testCase": "Mission type Special backward from Torcy to Poissy",
    "destination": Object {
      "letter": "T",
      "name": "Poissy",
      "order": 5,
      "section": "A5",
      "slug": "poissy",
    },
    "missionCode": "TOZZ",
    "origin": Object {
      "letter": "O",
      "name": "Torcy",
      "order": 25,
      "section": "A4",
      "slug": "torcy",
    },
    "type": 2,
    "way": "backward",
  },
  "result": Array [
    Object {
      "mission": "TOZZ",
      "stations": Array [
        "torcy",
        "lognes",
        "noisiel",
        "noisy+champs",
        "noisy+le+grand+mont+d'est",
        "bry+sur+marne",
        "neuilly+plaisance",
        "val+de+fontenay",
        "vincennes",
        "nation",
        "gare+de+lyon",
        "chatelet+les+halles",
        "auber",
        "charles+de+gaulle+etoile",
        "grande+arche+la+defense",
        "nanterre+prefecture",
        "houilles",
        "sartrouville",
        "maisons+laffitte",
        "acheres+grand+cormier",
        "poissy",
      ],
    },
  ],
}
`;

exports[`SchedulesForMissions getSchedulesForMissions should get schedule for missions UPAC 1`] = `
Object {
  "mission": Object {
    "_testCase": "Mission type Standard backward from Marne la Vallee-Chessy to Cergy-Le-Haut",
    "destination": Object {
      "letter": "U",
      "name": "Cergy-Le-Haut",
      "order": 1,
      "section": "A3",
      "slug": "cergy+le+haut",
    },
    "missionCode": "UPAC",
    "origin": Object {
      "letter": "Q",
      "name": "Marne la Vallee-Chessy",
      "order": 28,
      "section": "A4",
      "slug": "marne+la+vallee+chessy",
    },
    "type": 1,
    "way": "backward",
  },
  "result": Array [
    Object {
      "mission": "UPAC",
      "stations": Array [
        "marne+la+vallee+chessy",
        "val+d'europe+serris",
        "bussy+st+georges",
        "torcy",
        "lognes",
        "noisiel",
        "noisy+champs",
        "noisy+le+grand+mont+d'est",
        "bry+sur+marne",
        "neuilly+plaisance",
        "val+de+fontenay",
        "vincennes",
        "nation",
        "gare+de+lyon",
        "chatelet+les+halles",
        "auber",
        "charles+de+gaulle+etoile",
        "grande+arche+la+defense",
        "nanterre+prefecture",
        "houilles",
        "sartrouville",
        "maisons+laffitte",
        "acheres+ville",
        "conflans+fin+d'oise",
        "neuville+universite",
        "cergy+prefecture",
        "cergy+st+christophe",
        "cergy+le+haut",
      ],
    },
  ],
}
`;

exports[`SchedulesForMissions getSchedulesForMissions should get schedule for missions YOZZ 1`] = `
Object {
  "mission": Object {
    "_testCase": "Mission type Special backward from Torcy to Rueil Malmaison",
    "destination": Object {
      "letter": "Y",
      "name": "Rueil Malmaison",
      "order": 7,
      "section": "A1",
      "slug": "rueil+malmaison",
    },
    "missionCode": "YOZZ",
    "origin": Object {
      "letter": "O",
      "name": "Torcy",
      "order": 25,
      "section": "A4",
      "slug": "torcy",
    },
    "type": 2,
    "way": "backward",
  },
  "result": Array [
    Object {
      "mission": "YOZZ",
      "stations": Array [
        "torcy",
        "lognes",
        "noisiel",
        "noisy+champs",
        "noisy+le+grand+mont+d'est",
        "bry+sur+marne",
        "neuilly+plaisance",
        "val+de+fontenay",
        "vincennes",
        "nation",
        "gare+de+lyon",
        "chatelet+les+halles",
        "auber",
        "charles+de+gaulle+etoile",
        "grande+arche+la+defense",
        "nanterre+prefecture",
        "nanterre+universite",
        "nanterre+ville",
        "rueil+malmaison",
      ],
    },
  ],
}
`;
